bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -102,6 +102,10 @@ type KubeadmConfigSpec struct {
 	// Ignition contains Ignition specific configuration.
 	// +optional
 	Ignition *IgnitionSpec `json:"ignition,omitempty"`
+
+	//adminPass is the admin pass to inject to the machine
+	// +optional
+	AdminPass string `json:"adminPass"`
```

Comment:
If I'm not wrong, according to https://cloudinit.readthedocs.io/en/latest/topics/examples.html#including-users-and-groups this is already doable with https://github.com/kubernetes-sigs/cluster-api/blob/895b82ffeb0c3beff5ed21d97c23cfcd7abf22d2/bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go#L282

Have you considered this option?
/hold

Also, I have some concerns about this approach given that it requires to specify the admin password in plain text into the KubeadmConfig object, and then the same password will be repeated without encryption into the generated data secret that gets passed to machines.

@randomvariable opinions?

Reply:
After testing, we can use user group to update the password [.]( https://cloudinit.readthedocs.io/en/latest/topics/examples.html#including-users-and-groups)，and we can put the hashed passwd in the kubeadmConfig.If It is the right way ,I will continue to modify this pr

Reply:
not sure I got your comment. if you can use users, then no changes are required, or am I missing something?

Reply:
There need some changes, As the cloudinit remind(and verified by my test), we should use plain_text_passwd or hashed_passwd for already users.For simplicity and uniformity in coding, we can use hashed_passwd.

![cloudinit](https://user-images.githubusercontent.com/79310393/147797784-76f7d608-618e-4c5f-ace3-50c44771c189.png)

Reply:
Yes，there is no need any additional fields

Reply:
Just an additional data point, maybe it helps. 

CAPO uses the existing `passwd` and `lockPassword` fields to achieve setting a password for the user:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/test/e2e/data/ci-artifacts-platform-kustomization.yaml#L21-L33

Not sure if I follow the discussion correctly. Is your use case different / do the existing fields don't work in your case?

Reply:
This only suit for new users not for already users

Reply:
> not sure I got your comment. if you can use users, then no changes are required, or am I missing something?

There are some changes here. Please re-check,Thanks.

Reply:
Ah got it.

Imho it should be fine to add the new `HashedPasswd` field, but we cannot just drop the `Passwd` field from the v1beta1 API as it's a breaking change. So I would suggest to modify the PR to keep the `Passwd` field.

Reply:
> Ah got it.
> 
> In my opinion it should be fine to add the new `HashedPasswd` field, but we cannot just drop the `Passwd` field from the v1beta1 API as it's a breaking change. So I would suggest to modify the PR to keep the `Passwd` field.

so you mean we keep both` Passwd` and `HashedPasswd` fileds